src/js/game/Entities/Agent.js

```diff
@@ -94,6 +94,49 @@ module.exports = class Agent extends BaseEntity {
     this.collectItems();
   }
 
+  doMoveBackward(commandQueueItem) {
+    var player = this,
+      groundType,
+      jumpOff,
+      levelModel = this.controller.levelModel,
+      levelView = this.controller.levelView;
+    let wasOnBlock = player.isOnBlock;
+    let prevPosition = this.position;
+    // update position
+    levelModel.moveBackward(this);
+    // TODO: check for Lava, Creeper, water => play approp animation & call commandQueueItem.failed()
+
+    jumpOff = wasOnBlock && wasOnBlock !== player.isOnBlock;
+    if (player.isOnBlock || jumpOff) {
+      groundType = levelModel.actionPlane.getBlockAt(player.position).blockType;
+    } else {
+      groundType = levelModel.actionPlane.getBlockAt(player.position).blockType;
+    }
+
+    levelView.playMoveForwardAnimation(player, prevPosition, player.facing, jumpOff, player.isOnBlock, groundType, () => {
```

Comment:
since this method is being repurposed to be the animation for _all_ movement, it should also be renamed as such

Reply:
That's an interesting point. Let me talk to Joel. The Agent has the same move animation regardless of the direction, but I think that Alex/Steve actually DO have slightly different animations if they're walking backwards. For future proofing, if we ever think Alex/Steve would use different animations, it might be worth it to split these up ahead of time?

Reply:
oh, even better! Yeah, let's go ahead and split these up